src/components/panel/panel.js

```diff
@@ -355,6 +384,24 @@ angular
  * @param {!MdPanelPosition} position
  */
 
+/**
+ * @ngdoc method
+ * @name MdPanelRef#addToGroup
+ * @description
+ * Adds a panel to a group. The group must be created ahead of time.
```

Comment:
You probably should mention that a panel can only be part of one group, and this should be enforced when adding a panel to a group. (Although I'm not a fan of this decision, it's fine for an initial version.)

All of your logic is based around config.groupName, which is a string, not an array of strings.